src/gui/sharemanager.cpp

```diff
@@ -536,13 +536,21 @@ void ShareManager::slotShareCreated(const QJsonDocument &reply)
 
 void ShareManager::fetchShares(const QString &path)
 {
-    auto *job = new OcsShareJob(_account);
+    auto job = new OcsShareJob(_account);
     connect(job, &OcsShareJob::shareJobFinished, this, &ShareManager::slotSharesFetched);
     connect(job, &OcsJob::ocsError, this, &ShareManager::slotOcsError);
     job->getShares(path);
 }
 
-void ShareManager::slotSharesFetched(const QJsonDocument &reply)
+void ShareManager::fetchSharedWithMe(const QString &path)
+{
+    auto sharedWithMeJob = new OcsShareJob(_account);
```

Comment:
```suggestion
    const auto sharedWithMeJob = new OcsShareJob(_account);
```

IMO it is good to use const auto with pointer types; you won't be declaring the variable to be a pointer to a const `OcsShareJob`, rather, a const pointer to `OcsShareJob`

so `const auto sharedWithMeJob = new OcsShareJob(_account);`
will be a `OcsShareJob *const sharedWithMeJob`
not `const OcsShareJob *sharedWithMeJob`

the issue you were seeing before was caused by `const auto * sharedWithMeJob = new OcsShareJob(_account);` being understood as `const OcsShareJob *sharedWithMeJob`

Reply:
ohhhh!  I get it now, thanks for the explanation :>